packages/java/parser-jvm-test-utils/src/main/java/com/vaadin/hilla/parser/testutils/JsonAssertions.java

```diff
@@ -11,7 +13,35 @@ public final class JsonAssertions {
 
     public static void assertEquals(Object expected, Object actual)
             throws JsonProcessingException {
-        Assertions.assertEquals(printer.pretty().writeAsString(expected),
-                printer.pretty().writeAsString(actual));
+        try {
+            String expectedJson = printer.writeAsString(expected);
+            String actualJson = printer.writeAsString(actual);
+
+            // Use JSONAssert for comparison - ignores whitespace and provides
+            // precise diffs
+            JSONAssert.assertEquals(expectedJson, actualJson,
+                    JSONCompareMode.STRICT);
+        } catch (JSONException | AssertionError e) {
+            // Enhance error message with more context around the differing
+            // field
+            String originalMessage = e.getMessage();
+            String enhancedMessage = enhanceErrorMessage(originalMessage,
+                    expected, actual);
+            throw new AssertionError(enhancedMessage, e);
+        }
+    }
+
+    private static String enhanceErrorMessage(String originalMessage,
+            Object expected, Object actual) throws JsonProcessingException {
```

Comment:
This method could directly take expected and actual as strings instead of doing an additional conversion that might be different from the one used in test.